remote/remote_state_s3.go

```diff
@@ -381,20 +390,205 @@ func createS3BucketIfNecessary(s3Client *s3.S3, config *ExtendedRemoteStateConfi
 	return nil
 }
 
+func updateS3BucketIfNecessary(s3Client *s3.S3, config *ExtendedRemoteStateConfigS3, terragruntOptions *options.TerragruntOptions) error {
+	if !DoesS3BucketExist(s3Client, &config.remoteStateConfigS3.Bucket) {
+		return errors.WithStackTrace(fmt.Errorf("remote state S3 bucket %s does not exist or you don't have permissions to access it", config.remoteStateConfigS3.Bucket))
+	}
+
+	needUpdate, configBucket, err := checkIfS3BucketNeedsUpdate(s3Client, config, terragruntOptions)
+	if err != nil {
+		return err
+	}
+
+	if !needUpdate {
+		terragruntOptions.Logger.Debug("S3 bucket is already up to date")
+		return nil
+	}
+
+	prompt := fmt.Sprintf("Remote state S3 bucket %s is out of date. Would you like Terragrunt to update it?", config.remoteStateConfigS3.Bucket)
+	shouldUpdateBucket, err := shell.PromptUserForYesNo(prompt, terragruntOptions)
+	if err != nil {
+		return err
+	}
+
+	if shouldUpdateBucket {
```

Comment:
Make sense to return if `shouldUpdateBucket` is false? it will reduce number of nested blocks

```
if !shouldUpdateBucket {
  return nil
}
```

Reply:
changed